cognite/client/data_classes/postgres_gateway/users.py

```diff
@@ -0,0 +1,148 @@
+
+    The created postgres user only has access to use foreign tables and cannot directly create tables users. To create
+    foreign tables use the Postgres Gateway Tables APIs
+
+    This is the write/request format of the user.
+
+    Args:
+        credentials (SessionCredentials | None): Credentials for authenticating towards CDF using a CDF session.
+
+    """
+
+    def __init__(self, credentials: SessionCredentials | None = None) -> None:
+        self.credentials = credentials
+
+    @classmethod
+    def _load(cls, resource: dict[str, Any], cognite_client: CogniteClient | None = None) -> Self:
+        return cls(
+            credentials=SessionCredentials._load(resource["credentials"], cognite_client)
+            if "credentials" in resource
+            else None,
+        )
+
+    def dump(self, camel_case: bool = True) -> dict[str, Any]:
+        output = super().dump(camel_case=camel_case)
+        if isinstance(self.credentials, SessionCredentials):
+            output["credentials"] = self.credentials.dump(camel_case=camel_case)
+
+        return output
+
+    def as_write(self) -> UserWrite:
+        return self
+
+
+class User(_UserCore):
+    """A user.
+
+    This is the read/response format of the user.
+
+    Args:
+        username (str): Username to authenticate the user on the DB.
+        created_time (int): The number of milliseconds since 00:00:00 Thursday, 1 January 1970, Coordinated Universal Time (UTC), minus leap seconds.
+        last_updated_time (int): The number of milliseconds since 00:00:00 Thursday, 1 January 1970, Coordinated Universal Time (UTC), minus leap seconds.
+        session_id (int): ID of the session tied to this user.
+
+    """
+
+    def __init__(self, username: str, created_time: int, last_updated_time: int, session_id: int) -> None:
+        self.username = username
+        self.created_time = created_time
+        self.last_updated_time = last_updated_time
+        self.session_id = session_id
+
+    @classmethod
+    def _load(cls, resource: dict[str, Any], cognite_client: CogniteClient | None = None) -> Self:
+        return cls(
+            username=resource["username"],
+            created_time=resource["createdTime"],
+            last_updated_time=resource["lastUpdatedTime"],
+            session_id=resource["sessionId"],
+        )
+
+    def as_write(self) -> NoReturn:
+        raise TypeError(f"{type(self).__name__} cannot be converted to a write object")
+
+
+class UserUpdate(CogniteUpdate):
+    def __init__(
+        self,
+        username: str,
+    ) -> None:
+        super().__init__()
+        self.username = username
+
+    class _UpdateItemSessionCredentialsUpdate(CognitePrimitiveUpdate):
+        def set(self, value: SessionCredentials | None) -> UserUpdate:
+            return self._set(value.dump() if isinstance(value, SessionCredentials) else value)
+
+    @property
+    def credentials(self) -> UserUpdate._UpdateItemSessionCredentialsUpdate:
```

Comment:
A public method should not be returning private classes. So need to make `_UpdateItemSessionCredentialsUpdate` public
```suggestion
    def credentials(self) -> UserUpdate._UpdateItemSessionCredentialsUpdate:
```

Reply:
This is the pattern we use in all Update classes. The argument for keeping it is that making `_UpdateItemSessionCredentialsUpdate` public will clutter the available methods for the users, and then you return the private class you are expected to call `set` on it which will return a `UserUpdate` object.

Reply:
Ah, I missed that this was in an update class. That's fine 👍